rascal-vscode-extension/src/RascalExtension.ts

```diff
@@ -99,6 +101,30 @@ export class RascalExtension implements vscode.Disposable {
         );
     }
 
+
+    private registerCopyLocation() {
+        this.context.subscriptions.push(
+            vscode.commands.registerTextEditorCommand("rascalmpl.copyLocation", (text, _edit, moduleName) => {
+                const editor = vscode.window.activeTextEditor;
+                if (editor) {
+                    const fileName = editor.document.uri.fsPath;
+                    const selection = editor.selection;
+
+                    let offset = editor.document.offsetAt(selection.start);
+                    let length = editor.document.getText(selection).length;
+                    let startLine = selection.start.line;
+                    let startColumn = selection.start.character;
+                    let endLine = selection.end.line;
+                    let endColumn = selection.end.character;
+
+                    const location = `|file://${fileName}|(${offset},${length},<${startLine+1},${startColumn}>,<${endLine+1},${endColumn}>)`;
```

Comment:
not all uri's are file paths, the scheme and autority can be relevant.

there are some small differences between vscode uri print and rascal's uri print, but I think something like: `|${uri.toString()|` should work.